neo4j-jdbc/src/main/java/org/neo4j/jdbc/Neo4jConversions.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2023-2024 "Neo4j,"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.jdbc;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.OffsetTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.Objects;
+
+import org.neo4j.jdbc.values.Type;
+import org.neo4j.jdbc.values.Value;
+import org.neo4j.jdbc.values.Values;
+
+/**
+ * Various mappings from old to new Cypher types and Cypher types to JDBC types,
+ * additionally some more complex conversions and mappings.
+ *
+ * @author Neo4j Drivers Team
+ * @since 6.0.0
+ */
+final class Neo4jConversions {
+
+	private Neo4jConversions() {
+	}
+
+	/*
+	 * This is required because of a bug in 5 where java types are returned from the
+	 * db.schema.nodeTypeProperties() procedure In 6 this will change so will need to add
+	 * another method to handle the new types returned here when working with 6
```

Comment:
```suggestion
	 * This is required because of a bug in Neo4j 5 where java types are returned from the
	 * db.schema.nodeTypeProperties() procedure in Neo4j 6 this will change so will need to add
	 * another method to handle the new types returned here when working with Neo4j 6
```